### PR TITLE
Remove net5.0 from everywhere

### PR DESCRIPTION
**Pull request by cijothomas**
.NET 5.0 is EOL, so removing it from tests and CI to avoid wasting CI cycles.

Only product change is in OpenTelemetry.Extensions - changelog for this project is updated to reflect this change. For other projects, this is only removal of test targets.

### CODE REVIEW

**Review by Kielek — APPROVED** [review]
If you are making cleanup in old frameworks, please check also #580

**Comment by Kielek** [comment]
Also src\OpenTelemetry.Extensions\.publicApi\net5.0\* should be removed.
Please check if there is no more leftovers

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by cijothomas** [comment]
[xUnit.net 00:00:02.56]     OpenTelemetry.Instrumentation.AspNet.Tests.HttpInMetricsListenerTests.HttpDurationMetricIsEmitted [FAIL]
  Failed OpenTelemetry.Instrumentation.AspNet.Tests.HttpInMetricsListenerTests.HttpDurationMetricIsEmitted [419 ms]
  Error Message:
   Assert.Equal() Failure
Expected: 9.2037
Actual:   16.0556
  Stack Trace:
     at OpenTelemetry.Instrumentation.AspNet.Tests.HttpInMetricsListenerTests.HttpDurationMetricIsEmitted() in D:\a\opentelemetry-dotnet-contrib\opentelemetry-dotnet-contrib\test\OpenTelemetry.Instrumentation.AspNet.Tests\HttpInMetricsListenerTests.cs:line 79

Attachments:
  D:\a\opentelemetry-dotnet-contrib\opentelemetry-dotnet-contrib\TestResults\5decfa35-d491-46a3-9b5b-5a5ba0af9548\runneradmin_fv-az206-637_2022-08-31.17_23_05.coverage
Failed!  - Failed:     1, Passed:    21, Skipped:     0, Total:    22, Duration: 594 ms - OpenTelemetry.Instrumentation.AspNet.Tests.dll (net462)


@vishweshbankwar Can you see how this failure can occur!

File: src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // limitations under the License.
 // </copyright>
 
-#if NET462_OR_GREATER || NETSTANDARD2_0 || NET5_0_OR_GREATER
+#if NET462_OR_GREATER || NETSTANDARD2_0 || NET6_0_OR_GREATER
```
Comment (alanwest) [inline src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs:17]: 😕 When isn't this true?

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/2dcc721429779658f4b63ee140aad6361924fd79/src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj#L4

File: src/OpenTelemetry.Extensions/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@
 * Removes .NET Framework 4.6.1. The minimum .NET Framework version
   supported is .NET 4.6.2.
 
+  * Removes net5.0 target as .NET 5.0 is going out
+  of support. The package keeps netstandard2.0 target, so it
+  can still be used with .NET5.0 apps.
```
Comment (alanwest) [inline src/OpenTelemetry.Extensions/CHANGELOG.md:12]: This is not without its risks - though perhaps minor 🤞.

Here's an example of a potential consequence https://github.com/open-telemetry/opentelemetry-dotnet/pull/3222#discussion_r921392854. We removed `net5.0` target from the Zipkin exporter and now .NET 5 apps loose out on an optimization.

**Review by alanwest — APPROVED** [review]
LGTM with a heads up on potential (but maybe low) risk.

Also I suspect the compiler directives can be removed altogether from the OpenTelemetry.Extensions project.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs:17]: we have net6.0 in debug build. (it was net5.0)

**Review by mic-max — APPROVED** [review]

File: src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs (hunk 1 above)
Comment (alanwest) [reply src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs:17]: Though I think the directive is always `true`. #619

**Pull request merged**